homework/list.js

```diff
@@ -1,79 +1,142 @@
+        },
+        /**
+         * Dispatch a custom event when mouse click at the note-title.
+         * Also determine the content to be showed.
+         *
+         * @param {Object} event- mouse click event.
+         * @this {TodoListManager}
+         */
+        onNoteOpen(event) {
+            if (event.target.classList.contains('note-title')) {
+                var id = event.target.dataset.noteId;
+                var content = this._listNoteContent[id];
+                window.dispatchEvent(new CustomEvent('note-open',
+                  { detail: content }));
+            };
+        },
+        /** 
+         * Handle click event.
+         * If users click on note-title, call onNoteOpen to handle.
+         *
+         * @param {Object} event - mouse click event
+         * @this {TodoListManager}
+         */
+        handleEvent(event) {
+            switch (event.type) {
+                case 'click':
+                    this.onNoteOpen(event);
+                    break;
+            }
+        },
+        /**
+         * Show the Note for start stage.
+         *
+         * @this {TodoListManager}
+         */
+        preloadFirstNote() {
+            if (this._listNoteContent.length !== 0) {
+                var content = this._listNoteContent[0];
+                window.dispatchEvent(new CustomEvent('note-open',
+                  { detail: content }));
+            }
+        },
+        /**
+         * Update List.
+         *
+         * @param {array} - list to be updated to _listNoteContent
+         * @this {TodoListManager}
+         */
+        updateList(list) {
+            this._listNoteContent = list;
+        },
+        /**
+         * Draw the list.
+         *
+         * @this {TodoListManager}
+         */
+        drawList() {
+            var list = this._listNoteContent;
+            var ul = document.createElement('ul');
+            ul.id = 'note-title-list';
+            var buff = document.createDocumentFragment();
+            list.forEach(function (note, i) {
+                var li = document.createElement('li');
+                li.dataset.noteId = i;
+                li.classList.add('note-title');
+                li.textContent = note.title;
+                // Note: buff is captured, so we now have a
+                // little closure naturally.
+                buff.appendChild(li);
+            });
+            ul.appendChild(buff);
+            this._wrapper.appendChild(ul);
+        },
+        /**
+         * Try fetch Data(note-list) from .json,where is localhost.
+         * If there is any problem fetch the file direct from same http server as .html file
+         * (Because I got some problem to get the file from my system)
+         * If success,then continue the following flow 
+         * Else it will reject
+         *
+         * @return {Promise}
+         * @this {TodoListManager}
+         */
+        fetchList() {
+            return new Promise((function (yes, no) {
```

Comment:
Well, if you work along this may not be a problem. But to keep the documented names of arguments make your code more readable for others. Anyway, you have successfully "Promisified" the method, good work.